polkadot/runtime/common/src/xcm_sender.rs

```diff
@@ -138,6 +138,13 @@ where
 			.map(|()| hash)
 			.map_err(|_| SendError::Transport(&"Error placing into DMP queue"))
 	}
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn ensure_successful_delivery(location: Option<Location>) {
+		if let Some((0, [Parachain(id)])) = location.as_ref().map(|l| l.unpack()) {
+			dmp::Pallet::<T>::make_parachain_reachable(*id);
+		}
+	}
```

Comment:
nice :+1:, I think we could now add this to `ParentAsUmp`, `XcmpQueue` and maybe it will simplify and deduplicate other stuff - I can take a look as a follow-up